packages/editor/src/components/entities-saved-states/index.js

```diff
@@ -0,0 +1,97 @@
+/**
+ * External dependencies
+ */
+import { startCase } from 'lodash';
+
+/**
+ * WordPress dependencies
+ */
+import { Button, CheckboxControl, Modal } from '@wordpress/components';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { useState, useCallback } from '@wordpress/element';
+
+const EntitiesSavedStatesCheckbox = ( {
+	id,
+	name,
+	changes: { rawRecord },
+	checked,
+	setCheckedById,
+} ) => (
+	<CheckboxControl
+		label={ `${ startCase( name ) }: "${ rawRecord.title || rawRecord.name }"` }
+		checked={ checked }
+		onChange={ useCallback( ( nextChecked ) => setCheckedById( id, nextChecked ), [ id ] ) }
+	/>
+);
+
+export default function EntitiesSavedStates( { isOpen, onRequestClose } ) {
+	const entityRecordChangesByRecord = useSelect( ( select ) =>
```

Comment:
If I understand properly, we will be receiving changes from all entities here even entities that are not edited in the editor itself (think plugin, left over from other pages if it's an SPA...).

Do you think the "editor" should register the entities it's editing somehow instead of just getting all edits from the core data store?

Reply:
>If I understand properly, we will be receiving changes from all entities here even entities that are not edited in the editor itself (think plugin, left over from other pages if it's an SPA...).

Yes, I thought that wouldn't be a blocker since we only have one editor for now.

>Do you think the "editor" should register the entities it's editing somehow instead of just getting all edits from the core data store?

Yes, I think the `EntityProvider` should trigger an action that adds the provided entity record to a list in the editor store on mount. The editor can then use this list to query/filter entity changes.

Should I do that in this PR?

Reply:
I'm not sure it's the responsibility of `EntityProvider ` to do that as EntityProvider is not editor specific either.

I'm ok for this being in a separate PR.

Reply:
No, but it is meant to be rendered in an editing context and it's the only way to link the two dynamically without walking the block tree.

I.e. The only way for the editor to answer: Which entities am I rendering/editing?

Reply:
I was imagining something like the EditorProvider registering its entities and maybe the Post block, Site block registering their own. I can see how it's not great either as extra work for block authors. We can try and iterate.

Reply:
Yeah, and we can do it in `EntityProvider` in a way that does not break usage outside of an editor.